Feature `header-hide-controls-button`

CoroNaut opened the pull request:
Sometimes the context controls just add additional space when you don't use them anyway. This PR makes a button in the header of every buffer with context controls that is a little eye. Click that and your context controls will vanish or reappear. The state is saved in user data just like `minimize-headers`.

I put this feature beside the other two header buttons in `BASIC` features.

Before and after pictures:
<details>

![Capture](https://github.com/user-attachments/assets/0e9b336c-e9ef-4cb7-a41f-524de7cfad97)
![Capture2](https://github.com/user-attachments/assets/1ff4b990-f4e4-4e3b-9fb0-8b7833b21ca7)

</details>

Razenpok commented:
How about this:
- When the context bar is visible, it has a crossed-eye button inside the context bar. You can create a gap by creating a `flex-grow: 1` element between buttons)

<img width="459" alt="image" src="https://github.com/user-attachments/assets/d048086a-e523-4018-9971-dbde379526a2" />

- When the context bar is hidden, you show a bookmark button or some other icon that fits the meaning of the button

https://fontawesome.com/icons/bookmark?f=classic&s=solid
<img width="456" alt="image" src="https://github.com/user-attachments/assets/c733fff1-8e5d-4bfb-8054-9a2ec129fd19" />


This way, the feature won't clutter tile controls when the context bar is not hidden, which will likely be the case for most tiles.

Razenpok commented on src/features/basic/header-hide-controls-button/header-hide-controls-button.ts:
```diff
@@ -0,0 +1,38 @@
+
+async function onTileReady(tile: PrunTile) {
+  const tileContextControls = await $(tile.frame, C.ContextControls.container);
+  if (!tileContextControls) {
```
This check is unnecessary since `await $` will always produce non-nullish values

Razenpok commented on src/features/basic/header-hide-controls-button/header-hide-controls-button.ts:
```diff
@@ -0,0 +1,38 @@
+
+  const isMinimized = computedTileState(getTileState(tile), 'minimizeContextControls', false);
+  watchEffectWhileNodeAlive(tile.anchor, () => {
+    if (isMinimized.value) {
+      tileContextControls.style.display = 'none';
+    } else {
+      tileContextControls.style.display = 'flex';
+    }
```
Replace with addition and removal of the `css.hidden` class from utils

Razenpok commented on src/features/basic/header-hide-controls-button/tile-state.ts:
Merge this file into the feature file. The feature file is small enough, so it's not a problem to have this code there.

Razenpok commented:
<img width="343" alt="image" src="https://github.com/user-attachments/assets/8c1da276-07c3-4e54-b3ef-936490ac8bc5" />

Context controls look funny now

CoroNaut commented:
I changed up the context controls file as a first attempt at solving the problem, but then decided against it and forgot to revert the file. I do have it making a separate component in the PR and I reverted that file.

Razenpok commented:
Made some more changes:
1. Removed `css.HIDDEN` in favor of `isMinimized.value ? (TileControlsButton) : null`. You can conditionally render elements like this if writing JSX/TSX.
2. Replaced unicode characters with unicode values. In the future, always prefer unicode values, as they are easier to search for.
3. Inlined the `flex-grow: 1` div into the `createFragmentApp` call. Notice the `<>` syntax - that's a template. You can read about it here: https://vuejs.org/api/built-in-special-elements.html#template. In .tsx files they are written as `<></>`
